### ISSUE TITLE
optimize to obtain the maximum of a well trained neural network output 

### ISSUE THREAD

**Issue by AmosJoseph** [pytorch/botorch#1870]
Obtain the maximum of the output of a well trained neural network in the domain. 

I trained a neural network and tested it with good accuracy. 
Based on the built neural network, how to obtain the maximum of the output of it in the domain? 

Is there any such example? 
Could you please provide a minimal code example?

**Comment by Balandat** [comment]
If I understand this correctly you are suggesting you want to use Bayesian Optimization to find `max_{x \in X} NN(x)`, where `X` is your domain? How expensive is it to evaluate the `NN`? Typically a NN forward pass will be very cheap and so Bayesian Optimization is likely not a great fit for your problem, you'll likely want to look at standard numerical optimization algorithms.

Closing this out since it doesn't seem very relevant to BoTorch, but please feel free to re-open if I misunderstood your question.

**Comment by AmosJoseph** [comment]
Yes, the built NN is very cheap to evaluate. 
And Bayesian Optimization is good enough and capable of this task with fewer iteration.

There are many tutorials about optimization of NN when trainng a NN.
But optimizing the output of trained NN is scarce.

**Comment by eytan** [comment]
you can just use standard gradient-based optimization (e.g., run adam
multiple times until convergence from a few random initialization, and pick
the max) to optimize the input parameters.  I am not sure how bounds are
typically expressed in these cases but one hacky solution is to stick some
sigmoids at the bounds of your search space to smoothly spike the objective
to some large negative value.
